curated_transformers/models/attention.py

```diff
@@ -224,18 +237,40 @@ def __init__(
             dropout_prob=dropout_prob,
         )
 
+        if (
+            qkv_mode == QkvMode.MERGED_SPLIT_BEFORE
+            and qkv_head_sharing == QkvHeadSharing.KEY_VALUE
+        ):
+            raise ValueError(
+                "QkvMode.MERGED_SPLIT_BEFORE is incompatible with key/value head sharing"
+            )
+        self.head_sharing = qkv_head_sharing
+
+        # Head sharing is implemented as just having one head. We will still
+        # have multiple heads after attention, because the value has multiple
+        # heads, so attention will broadcast.
+        kv_dim = (
+            self.dims_per_head
+            if qkv_head_sharing == QkvHeadSharing.KEY_VALUE
+            else self.model_dim
+        )
         if qkv_mode == QkvMode.SEPARATE:
-            self.query = Linear(self.model_dim, self.model_dim, device=device)
-            self.key = Linear(self.model_dim, self.model_dim, device=device)
-            self.value = Linear(self.model_dim, self.model_dim, device=device)
+            self.query = Linear(
+                self.model_dim, self.model_dim, bias=use_bias, device=device
+            )
+            self.key = Linear(self.model_dim, kv_dim, bias=use_bias, device=device)
+            self.value = Linear(self.model_dim, kv_dim, bias=use_bias, device=device)
         else:
             self.input = Linear(
                 self.model_dim,
-                self.model_dim * 3,
+                self.model_dim + 2 * kv_dim,
+                bias=use_bias,
```

Comment:
Was the absence of `bias` an oversight until now or is it required for Falcon?

Reply:
`bias` is enabled by default. Falcon is the first model that we support that doesn't use bias for linear layers.